### ISSUE TITLE
pipeline number larger or equel to 4 will encounter Child process setup error

### ISSUE THREAD

**Issue by guoh27** [gmarcais/NoShell#1]
As title says, for example, dd if=a | dd | dd | dd of=b, will fail.

**Comment by SplitGemini** [comment]
error show bad file descriptor

modified as below works, but I don't know why

noshell.cc 
```cpp
Exit PipeLine::run()
...
std::set<int> r;
  for(++it; it != commands.end(); pit = it, ++it) {
    int fds[2];
    if(pipe2(fds, O_CLOEXEC) == -1) exit(1); // TODO: handle error
    fix_collision(fds[0], r);
    fix_collision(fds[1], r);
    r.insert(fds[1]);
    r.insert(fds[0]);
...
```

**Comment by gmarcais** [comment]
Let me give it a look. I am not quite sure why it would do that.

**Comment by gmarcais** [comment]
Fixed in 0.2.0
